File: baekjoon/1697.py
import collections
N,K=map(int,input().split())
def BFS():
    visit = {}
    frontier=collections.deque()
    frontier.append((N,0))
    while frontier:
        u=frontier.popleft()
        for v in [u[0]-1,u[0]+1,2*u[0]]:
            if v==K:
                return u[1]+1
            if 0<=v<=100001 and v not in visit:
                visit[v]=True
                frontier.append((v,u[1]+1))
if N==K:
    print(0)
else:
    print(BFS())

File: baekjoon/1697.cpp
#include <iostream>
#include <deque>
#include <map>
using namespace std;
int N, K;
deque<int> frontier;
bool visited[200000 + 3] = {false};
int BFS()
{
    if (N == K)
        return 0;
    visited[N] = true;
    frontier.push_back(N);
    int time = 1;
    while (frontier.size())
    {
        deque<int> next;
        for (deque<int>::iterator it = frontier.begin(); it != frontier.end(); it++)
        {
            if (*it + 1 == K || *it - 1 == K || *it * 2 == K)
                return time;
            if (*it + 1 <= 100000 && !visited[*it + 1])
            {
                visited[*it + 1] = true;
                next.push_back(*it + 1);
            }
            if (*it - 1 >= 0 && !visited[*it - 1])
            {
                visited[*it - 1] = true;
                next.push_back(*it - 1);
            }
            if (*it * 2 <= 100000 && !visited[*it * 2])
            {
                visited[*it * 2] = true;
                next.push_back(*it * 2);
            }
        }
        frontier = next;
        time++;
    }

    return -1;
}
int main(void)
{
    cin >> N >> K;
    cout << BFS();
    return 0;
}